### PR TITLE
Don't serve unlisted versions when checking for add-on updates

### PR DESCRIPTION
**Pull request by diox**
Part of #3567

### CODE REVIEW

File: src/olympia/addons/tests/test_update.py [hunk 1]
```diff
@@ -91,12 +91,14 @@ def test_no_version(self):
         assert up.is_valid()
 
     def test_unlisted_addon(self):
-        """Don't provide updates for unlisted addons."""
+        """Add-ons with only unlisted versions are valid, they just don't
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_update.py]: I know this is current behaviour, but should an Addon with no valid versions (e.g. all deleted; all unlisted; etc) be valid?

Comment (diox) [inline src/olympia/addons/tests/test_update.py]: It will be valid, as long as it has not been deleted or disabled by the author(s).

Comment (diox) [inline src/olympia/addons/tests/test_update.py]: In fact there is a test covering that - `test_no_version()` just above this one.

Comment (diox) [inline src/olympia/addons/tests/test_update.py]: Ah, I misunderstood :) I don't think it matters much, the code serves a different rdf if there are no versions. 

There is a related issue that's more worrying though: https://github.com/mozilla/addons-server/issues/3228 causes `is_valid()` to return `True` if the addon status is `STATUS_DISABLED`. I should change that, but in a different PR.

**Pull request merged**